### PR TITLE
tests: add tests with tagging and upload

### PR DESCRIPTION
**Pull request by murlock**
It requires this PR https://github.com/open-io/swift3/pull/92 before merging

### CODE REVIEW

**Review by AymericDu — APPROVED** [review]

File: tests/functional/s3-tagging.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    data = run_s3api("get-object-tagging", "--bucket", BUCKET,
+                     "--key", "magic")
+    assert data['TagSet'][0]['Value'] == tagging['unicode'].decode('utf8')
+
+    tagging = {'été': ''}
+    run_s3api("put-object", "--bucket", BUCKET, "--key", "magic",
+              "--tagging", urlencode(tagging))
+    data = run_s3api("get-object-tagging", "--bucket", BUCKET, "--key",
+                     "magic")
+    assert data['TagSet'][0]['Key'] == 'été'.decode('utf8')
```
Comment (fvennetier) [inline tests/functional/s3-tagging.py:45]: We cannot decode a string.
```
>>> 'été'.decode('utf8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'
```

**Review by fvennetier — COMMENTED** [review]
Is there any reason for these tests to be in a new script rather than the existing `s3-tagging.sh` script?

By the way, since there are several independent tests, I suggest to make several functions rather than a big one.

File: tests/functional/s3-tagging.py (hunk 1 above)
Comment (murlock) [reply tests/functional/s3-tagging.py:45]: Writting test in shell is very prone to error IMO and more difficult to split, I'll however split mine in small parts

My bad, I didn't tested with python3

File: tests/functional/s3-tagging.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def run_s3api(*params, **kwargs):
+    print(*params)
+    cmd = AWS + list(params)
+    try:
+        out = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        # some s3api commands return nothing and json.loads
+        # doesn't like it
+        if not len(out):
```
Comment (fvennetier) [inline tests/functional/s3-tagging.py:29]: `len()` is redundant. `if not out:` is sufficient. But that is not a release blocker. :slightly_smiling_face:

**Review by fvennetier — APPROVED** [review]

**Review by AymericDu — APPROVED** [review]

File: tests/functional/s3-tagging.py (hunk 2 above)
Comment (murlock) [reply tests/functional/s3-tagging.py]: Done

**Pull request merged**